modules/imgproc/src/pyramids.cpp

```diff
@@ -84,13 +84,36 @@ template<typename T1, typename T2> int PyrUpVecV(T1**, T2**, int) { return 0; }
 
 #if CV_SIMD
 
+template<typename T>
+#if CV_SIMD_WIDTH == 16
+using VecType = typename Type2Vec128_Traits<T>::vec_type;
+#elif CV_SIMD_WIDTH == 32
+using VecType = typename Type2Vec256_Traits<_T>::vec_type;
+#elif CV_SIMD_WIDTH == 64
+using VecType = typename Type2Vec512_Traits<_T>::vec_type;
+#else
+#error "Build configuration error, unsupported CV_SIMD_WIDTH"
+#endif
+
+// construct vector: {a, b, a, b, ... , a, b}
+template<typename T>
+static
+VecType<T> make_interleaved_vector(T a, T b)
+{
+    VecType<T> v_a = vx_setall<T>(a);
+    VecType<T> v_b = vx_setall<T>(b);
+    VecType<T> v_a_b, v_a_b_mirror;
+    v_zip(v_a, v_b, v_a_b, v_a_b_mirror);
+    return v_a_b;
+}
+
 template<> int PyrDownVecH<uchar, int, 1>(const uchar* src, int* row, int width)
 {
     int x = 0;
     const uchar *src01 = src, *src23 = src + 2, *src4 = src + 3;
 
-    v_int16 v_1_4 = v_reinterpret_as_s16(vx_setall_u32(0x00040001));
-    v_int16 v_6_4 = v_reinterpret_as_s16(vx_setall_u32(0x00040006));
```

Comment:
Lets reuse `#ifdef WORDS_BIGENDIAN` from here: https://github.com/opencv/opencv/blob/3.4.16/cmake/templates/cvconfig.h.in#L200

Example:
```
#ifndef WORDS_BIGENDIAN
    v_int16 v_1_4 = v_reinterpret_as_s16(vx_setall_u32(0x00040001));
#else
    v_int16 v_1_4 = v_reinterpret_as_s16(vx_setall_u32(0x00010004));
#endif
```